### DISCUSSION TITLE
Disable errors/hints for `anon` role

### DISCUSSION THREAD

**Discussion by bombillazo** [PostgREST/postgrest#3397] [Q&A] [answered]
Hello, I am looking for a way to disable any and all error/hints returns by the Rest API endpoint if the anon role is being used. I disabled the openapi_mode for `anon` with `ALTER ROLE anon SET pgrst.openapi_mode TO 'disabled';` but calling any REST API endpoint still returns DB errors that expose schema details.

<img width="724" alt="image" src="https://github.com/PostgREST/postgrest/assets/44002321/e2ada4f4-b382-4220-b877-ee4fad0fcc77">

**Comment by wolfgangwalther** [comment, marked as answer]
> I am looking for a way to disable any and all error/hints returns by the Rest API endpoint if the anon role is being used.

There is no way to do that.

In general we don't consider the structure of the exposed schema "secret".

**Reply by bombillazo** [reply]
Got it. We're using Supabase and the proper way to avoid this would have been not to use the `public` schema to begin with. It is one of those things you learn along the way much later after you've started. Migrating all my schema to a custom private one at this point will be too much of a hassle

**Reply by steve-chavez** [reply]
Might be possible to customize the error messages (or empty them) later on with https://github.com/PostgREST/postgrest/issues/2706

**Reply by bombillazo** [reply]
Nice, hope we get the ability to customize those outputs.
